tests/Oryx.Tests.Common/EndToEndTestHelper.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Oryx.Tests.Common
     public static class EndToEndTestHelper
     {
         private const int MaxRetryCount = 100;
-        private static readonly TimeSpan DelayBetweenRetries = TimeSpan.FromSeconds(100);
+        private static readonly TimeSpan DelayBetweenRetries = TimeSpan.FromSeconds(2);
```

Comment:
What does this timespan set up for? Does it reduce the retry waiting time when timeout occurs?

Reply:
Yeah I changed it a while back because this was very flaky: https://github.com/microsoft/Oryx/commit/0cd0b89843e4fcc32a3141fe77b503282e23747f#

I do wonder if 2 seconds is too little. Network ports usually seem to take a few seconds to retry (open and close)

Reply:
Could we bump this up from 2 seconds to 10 seconds (what it was originally)? My concern is the when perfroming a retry, the port that it's trying to close and re-open hasn't been closed completely and cause errors when trying to re-open. Unless we're sure 2 seconds will suffice

Reply:
Sure I can bump back to the original delay. 

@qianz2 this timespan is the delay in between each retry of trying to get the host port, and since we wait this long [before we try to get the host port for the first time](https://github.com/microsoft/Oryx/pull/1471/files/7d2d507ce5e559dc8e4c3b17f0457a42bf7aaec6#diff-98cee7e79c6b4790f7b445c7de5c47e1fc95b79b19941e94cf70a13013e10b0cR412), we were introducing a minimum of a 100 second delay before we even attempted to see if the container was up and had a port that we could access. We needed this delay in the past since it gave time (100 seconds) for the container to start up. One of the changes in this PR is to introduce a new method that waits for the container to start up (checking every few seconds) before we try to connect to the host port, so in theory we can then attempt to reconnect to the host port here every few seconds and since the container exists we will not error out on the first attempt.